tests/terminal.test.js

```diff
@@ -0,0 +1,70 @@
+const Terminal = require('../model/terminal.js');
+const Port = require('../model/port.js');
+const Component = require('../model/component.js');
+const Coord = require('../model/coord.js');
+
+// Suppress console logs
+console.log = jest.fn();
+
+var layer1 = 'layer-1';
+var layer2 = 'layer-2';
+var validPort1 = new Port('port-1', layer1, new Coord(0, 0));
+var validPort2 = new Port('port-2', layer2, new Coord(0, 5));
+var validPort3 = new Port('port-3', 'layer-3', new Coord(0, 10));
+var invalidPort = new Port('port-1', '', new Coord(0, 0));
+var validComponent = new Component('name', 'unique-id', [layer1, layer2], 10, 20, 'entity', [validPort1, validPort2]);
+var invalidComponent = new Component('', '', [layer1, layer1], 0, 0, '', [new Port(), new Port()]);
+
+test('initialize Terminal: parameters', () => {
+    let t = new Terminal(new Component(), new Port());
+
+    expect(t.component).toEqual(new Component());
+    expect(t.port).toEqual(new Port());
+});
+
+test('initialize Terminal: default', () => {
+    let t = new Terminal();
+
+    expect(t.component).toBe(null);
+    expect(t.port).toBe(null);
+});
+
+test('modify Terminal', () => {
+    let t = new Terminal();
+
+    t.component = new Component();
+    t.port = new Port();
+
+    expect(t.component).toEqual(new Component());
+    expect(t.port).toEqual(new Port());
```

Comment:
For this test and the one above you may want to also test that the terminal created isn't valid.

Reply:
I added two new tests to check that modified Terminals still validate correctly. The one above (initialize Terminal: default) has a validation test all the way at the bottom.